src/Config/Type.hs

```diff
@@ -119,11 +121,28 @@ instance Semigroup RestrictIdents where
     OnlyIdents x1 <> OnlyIdents x2 = OnlyIdents $ x1 <> x2
     ri1 <> ri2 = error $ "Incompatible restrictions: " ++ show (ri1, ri2)
 
+data RestrictImportStyle
+  = ImportStyleQualified
+  | ImportStyleUnqualified
+  | ImportStyleExplicit
+  | ImportStyleExplicitOrQualified
+  | ImportStyleIgnore
+  deriving Show
+
+data QualifiedStyle
+  = QualifiedStylePre
+  | QualifiedStylePost
+  | QualifiedStyleIgnore
```

Comment:
Ditto, QualifiedStyleUnrestricted?

Although I wonder how useful qualified style is? There is a GHC warning to require post qualified, so wouldn't people just use that? Although I'm not against it.

Reply:
As I outlined in the OP, the primary reason for having this is not as much to have a warning (as you say, GHC already has a warning for this), as to be able to specify the default for HLint's hints. It would be pretty annoying if HLint says to use `import qualified X` and then GHC complains that it should be `import X qualified` instead, I think. And since we already have the option, adding a check is literally two lines of code.